[new] create a fork using aiven cli (#1503)
timescale/docs: mst/aiven-client/create-fork.md

---
title: Create a fork using Aiven Client 
excerpt: Create a fork of the service using Aiven Command Line tool for fully managed services on AWS, Azure, or GCP.
product: mst
---

# Create a fork of your service using Aiven Client

When you a fork a service, you create an exact copy of the service, including
the underlying database. You can use it to create a development copy of your
production environment, set up a snapshot to analyze an issue or test an
upgrade, or create an instance in a different cloud, geographical location, or

> under a different plan.

Reviewer: Non-blocking suggestion: set this up as bullets, easier to skim

## Before you begin

*   [Install Aiven Client and connect][aiven-client-install] to Managed Service
    for TimescaleDB
*   [Create a service][create-service] in Managed Service for TimescaleDB

<procedure>

## Creating a fork of your service

1.  Switch to the `Project` that contains the service that you want to fork:

    ```bash
     avn switch <PROJECT>
    ```

1.  List the services in the `Project`:

    ```bash
     avn service list
    ```

1.  Make a note of the service that you want to fork, listed under
    `SERVICE_NAME` column in the output.

1.  Get the details of the service that you want to fork:

    ```bash
    avn service get <SERVICE_NAME>
    ```

1.  Create a fork of the service

    ```bash

>     avn service create <NAME_OF_FORK> --project <PROJECT> -t <SERVICE_TYPE> --plan <PLAN> --cloud <CLOUD_NAME> -c service_to_fork_from=<NAME_OF_SERVICE_TO_FORK>

Reviewer: Non-blocking suggestion: split this onto multiple lines with line continuation (`\`) because otherwise it's quite a bit of scrolling to read the whole thing

    ```

    For more information about projects, plans, and other details about
    services,see [Services][about-mst].

</procedure>

## Example

To create a fork named `grafana-fork` for a service named `grafana` in `fork-project`

> on `timescale-aws-us-east-1` with the plan `dashboard-1`: 

Reviewer: Non-blocking suggestion: This is hard to parse because prepositional phrase galore! Maybe try splitting into multiple sentences?

> ```bash

Reviewer: Same comment about splitting onto multiple lines

   avn service create grafana-fork --project project-fork -t grafana --plan dashboard-1 --cloud timescale-aws-us-east-1  -c service_to_fork_from=grafana
```

You can switch to `project-fork` and view the newly created `grafana-fork` using:

```bash
   avn service list
```

[about-mst]: /mst/:currentVersion:/about-mst/

> [aiven-client-install]: /mst/aiven-client/aiven-client-install/

Reviewer: ```suggestion
[aiven-client-install]: /mst/:currentVersion:/aiven-client/aiven-client-install/
```

---
title: Create a fork using Aiven Client 
excerpt: Create a fork of the service using Aiven Command Line tool for fully managed services on AWS, Azure, or GCP.
product: mst
---

> # Create a fork of your service using Aiven Client

Reviewer 2: ```suggestion
# Fork services with Aiven Client
```

When you a fork a service, you create an exact copy of the service, including
the underlying database. You can use a fork of your service to:

> *   create a development copy of your production environment
> *   set up a snapshot to analyze an issue or test an upgrade
> *   create an instance in a different cloud, geographical location, or under
>     a different plan.

Reviewer 2: ```suggestion

*   Create a development copy of your production environment.
*   Set up a snapshot to analyze an issue or test an upgrade.
*   Create an instance in a different cloud, geographical location, or under
    a different plan.
```

## Before you begin

> 
> *   [Install Aiven Client and connect][aiven-client-install] to Managed Service
>     for TimescaleDB
> *   [Create a service][create-service] in Managed Service for TimescaleDB

Reviewer 2: ```suggestion

Before you begin, make sure you have: 

*   Installed [Aiven Client][aiven-client-install].
*   Signed in to your [Managed Service for TimescaleDB dashboard][mst-login].
*   Created a [service][create-service] in Managed Service for TimescaleDB.
```

<procedure>

## Creating a fork of your service

> 1.  Switch to the `Project` that contains the service that you want to fork:

Reviewer 2: ```suggestion
1.  In the Aiven Client, log in to your Managed Service for TimescaleDB account.

1.  Switch to the project that contains the service you want to fork:
```

    ```bash
     avn switch <PROJECT>
    ```

> 1.  List the services in the `Project`:

Reviewer 2: ```suggestion
1.  List the services in the project:
```

    ```bash
     avn service list
    ```

1.  Make a note of the service that you want to fork, listed under

>     `SERVICE_NAME` column in the output.

Reviewer 2: This is working with the output of the previous step, so these could be combined.

1.  Get the details of the service that you want to fork:

    ```bash
    avn service get <SERVICE_NAME>
    ```

> 1.  Create a fork of the service

Reviewer 2: ```suggestion
1.  Create the fork:
```

    ```bash
    avn service create <NAME_OF_FORK> --project <PROJECT>\
    -t <SERVICE_TYPE> --plan <PLAN> --cloud <CLOUD_NAME>\
    -c service_to_fork_from=<NAME_OF_SERVICE_TO_FORK>
    ```

    For more information about projects, plans, and other details about

>     services,see [Services][about-mst].

Reviewer 2: This feels a bit generic, I'm not sure it's required here. Perhaps if you feel it's important, it could go in the concept info at the top of the page instead?

</procedure>

## Example

To create a fork named `grafana-fork` for a service named `grafana` with these parameters:
* PROJECT: `fork-project`
* CLOUD_NAME: `timescale-aws-us-east-1`
* PLAN: `dashboard-1` 

```bash
   avn service create grafana-fork --project project-fork -t grafana --plan dashboard-1 --cloud timescale-aws-us-east-1  -c service_to_fork_from=grafana
```

You can switch to `project-fork` and view the newly created `grafana-fork` using:

```bash
   avn service list
```

[about-mst]: /mst/:currentVersion:/about-mst/
[aiven-client-install]: /mst/:currentVersion:/aiven-client/aiven-client-install/

> [create-service]: /install/:currentVersion:/installation-mst/#create-your-first-service

Reviewer 2: ```suggestion
[create-service]: /install/:currentVersion:/installation-mst/#create-your-first-service
[mst-login]: https://portal.managed.timescale.com
```

When you a fork a service, you create an exact copy of the service, including
the underlying database. You can use a fork of your service to:

*   Create a development copy of your production environment.
*   Set up a snapshot to analyze an issue or test an upgrade.
*   Create an instance in a different cloud, geographical location, or under
    a different plan.

## Before you begin

Before you begin, make sure you have: 

*   Installed [Aiven Client][aiven-client-install].
*   Signed in to your [Managed Service for TimescaleDB dashboard][mst-login].
*   Created a [service][create-service] in Managed Service for TimescaleDB.

<procedure>

## Creating a fork of your service

> 1.  In the Aiven Client, log in to your Managed Service for TimescaleDB account.

Reviewer 2: ```suggestion
1.  In the Aiven Client, log in to your [Managed Service for TimescaleDB dashboard][mst-login].
```
